### PR TITLE
Add OpenMPT 1.28.05.00

### PR DESCRIPTION
**Pull request by MunsterPlop**

### CODE REVIEW

**Comment by linsui** [comment]
Please add autoupdate.hash

File: bucket/openmpt.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+    "homepage": "https://openmpt.org/",
+    "description": "OpenMPT is a powerful audio application that makes writing music fun, easy and efficient.",
+    "license": "BSD-3-Clause",
+    "version": "1.28.05.00",
+    "architecture": {
+        "32bit": {
+            "url": "https://openmpt.org/download_file.php?file=OpenMPT-1.28.05.00-Setup.exe",
```
Comment (Ash258) [inline bucket/openmpt.json]: I would rather use direct URL.

```suggestion
            "url": "https://download.openmpt.org/OpenMPT-1.28.05.00-Setup.exe",
```

**Comment by MunsterPlop** [comment]
@linsui 

Sorry, I'm not exactly regex-savvy so I'm not too sure how to get the corresponding hash for each architecture. I suppose I need to grab from [https://openmpt.org/download](url) the 1st occurence of a md5 or sha1 hash for the 32-bit and the 2nd occurence for the 64-bit, but I'm not managing.

I also saw  there are .digests files on this page: [https://download.openmpt.org/](url) that could make it simpler. Is there any way to use these ?

File: bucket/openmpt.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            "hash": "07aaf66792dbe0d04f41cee842b639ab5959440f80e9683179aaafb86698bf6a"
+        },
+        "64bit": {
+            "url": "https://openmpt.org/download_file.php?file=OpenMPT-1.28.05.00-Setup-x64.exe",
```
Comment (Ash258) [inline bucket/openmpt.json]:
```suggestion
            "url": "https://download.openmpt.org/OpenMPT-1.28.05.00-Setup-x64.exe",
```

File: bucket/openmpt.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            "hash": "033c31cd9b20924bbb36cbe52a889c042b13448b8ca38302b5ca128417b2e0c2"
+        }
+    },
+    "innosetup": true,
+    "shortcuts": [
+        [
+            "mptrack,1.exe",
```
Comment (Ash258) [inline bucket/openmpt.json]: I will post fix to this later when I am on PC

File: bucket/openmpt.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        ]
+    ],
+    "checkver": {
+        "url": "https://openmpt.org/download",
+        "regex": "OpenMPT ([\\d.]+)"
+    },
+    "autoupdate": {
+        "architecture": {
+            "32bit": {
+                "url": "https://openmpt.org/download_file.php?file=OpenMPT-$version-Setup.exe"
```
Comment (Ash258) [inline bucket/openmpt.json]: Adopt direct URLS please

File: bucket/openmpt.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            },
+            "64bit": {
+                "url": "https://openmpt.org/download_file.php?file=OpenMPT-$version-Setup-x64.exe"
+            }
+        }
```
Comment (Ash258) [inline bucket/openmpt.json]: Something like this should work. Needs testing

```suggestion
        },
        "hash": {
            "url": "https://openmpt.org/download",
            "regex": "$basename.*?$sha1"
        }
```

Comment (Ash258) [reply bucket/openmpt.json]: digest file will be better

Comment (Ash258) [reply bucket/openmpt.json]: Just use `$url.digests` with regex `$sha1`

File: bucket/openmpt.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
+{
+    "homepage": "https://openmpt.org/",
+    "description": "OpenMPT is a powerful audio application that makes writing music fun, easy and efficient.",
```
Comment (Ash258) [inline bucket/openmpt.json]:
```suggestion
    "description": "Powerful audio application that makes writing music fun, easy and efficient.",
```

File: bucket/openmpt.json (hunk 3 above)
Comment (Ash258) [reply bucket/openmpt.json]:
```suggestion
            "mptrack.exe",
```

File: bucket/openmpt.json [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        "32bit": {
+            "url": "https://download.openmpt.org/OpenMPT-1.28.05.00-Setup.exe",
+            "hash": "sha1:990d42d44cc4da9b5720ee53509f53dd4b2c1e3f"
+        },
+        "64bit": {
+            "url": "https://download.openmpt.org/OpenMPT-1.28.05.00-Setup-x64.exe",
+            "hash": "sha1:9743268e1f5426087ace72dae50791a9c95051f1"
+        }
+    },
+    "innosetup": true,
```
Comment (Ash258) [inline bucket/openmpt.json]:
```suggestion
    "innosetup": true,
    "pre_install": [
        "Remove-Item \"$dir\\*,2*\" -Force",
        "Get-ChildItem \"$dir\" '*,1*' | ForEach-Object { Rename-Item $_.Fullname \"$($_.Basename.TrimEnd(',1'))$($_.Extension)\" }"
    ],
```

**Pull request merged**